app/models/permitted_params.rb

```diff
@@ -153,6 +153,10 @@ def project_type
     params.require(:project_type).permit(*self.class.permitted_attributes[:project_type])
   end
 
+  def projects_type_ids
+    params.require(:project).require(:type_ids).map(&:to_i).select { |x| x > 0 }
```

Comment:
It really doesn't matter here at all, but everytime I see `map+select` I'm reminded that `map+compact` will build no intermediary arrays and [performs better](http://stackoverflow.com/a/5046057).